File: frontend/src/app/core/services/school-class-subject/school-class-subject.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { SchoolClassSubjectDto } from '../../../task/model/school-class-subject.dto';
import { CacheableService } from '../core/cacheable.service';

@Injectable()
export class SchoolClassSubjectService extends CacheableService<Array<SchoolClassSubjectDto>> {
  constructor(http: HttpClient) {
    super(http, '/api/schoolClassSubject');
  }

  public getSchoolClassSubjects(): Observable<Array<SchoolClassSubjectDto>> {
    return this.getFromCacheOrFetch(() => this._http.get<Array<SchoolClassSubjectDto>>(this._baseUri));
  }
}

File: backend/src/main/kotlin/outcobra/server/service/SchoolClassSubjectService.kt
package outcobra.server.service

import outcobra.server.model.dto.SchoolClassSubjectDto
/**
 * @author Florian Bürgi
 * @since 1.2.0
 */
interface SchoolClassSubjectService {

    /**
     * @return a list of [SchoolClassSubjectDto]'s for the current user
     */
    fun getSchoolClassSubjects(): List<SchoolClassSubjectDto>
}